app/controllers/secrets_controller.rb

```diff
@@ -54,6 +54,26 @@ def duplicates
       sort_by { |_, v| -v.size }
   end
 
+  def resolve
+    params.require([:project_id, :deploy_group, :keys])
```

Comment:
doing the require on each line as they are used reduces duplication

Reply:
i don't understand this -- sorry 🤕

Reply:
yeah